libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -369,4 +380,14 @@ public void onServiceDisconnected(ComponentName className) {
       serviceBound = false;
     }
   };
+
+  @Override
+  public void onHttpResponse(Response response) {
```

Comment:
Why are `TelemetryListener` callbacks empty?

https://github.com/mapbox/mapbox-events-android/pull/38#discussion_r160396201
> Ultimately, registering to a listener should be optional and should not affect the overall behavior ensuring that everything is still working without any action from the client/user.

The main purpose of implementing a custom callback is to have more control over what we do internally and what we reply externally.
That is we should receive and manage the HTTP callback internally 👀 
> This could be achieved if `MapboxTelemetry` implements `Callback` and assigning `httpCallback` to `this`, avoiding to change any other classes other than `MapboxTelemetry`.

and then when `onFailure` and `onResponse` methods are fired, adapt the data that we're going to expose (not return directly what we receive from `Callback` 👀 https://github.com/mapbox/mapbox-java/blob/mas-2.x.x/mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryListener.java#L21) and notify the clients which are subscribed (call `TelemetryListener` methods wisely on the `telemetryListeners` registered).
> It'd be great though to add a custom callback which would give us more flexibility on what actions to notify (i.e. to notify actions beyond `onResponse` and `onFailure`), more control over what and when actions need to be notified and it'd open also the possibility of having multiple listeners that users/clients could add/remove as needed.

This way the only changes needed are in `MapboxTelemetry`.
We followed this approach in the old implementation.